gcp_variant_transforms/libs/bigquery_vcf_schema_converter.py

```diff
@@ -222,24 +215,73 @@ def _add_format_fields(schema, formats):
-    formats.update({
-        field.name: _Format(field.name,
+    if field.name in vcf_reserved_fields.FORMAT_FIELDS.keys():
+      _add_reserved_field(field,
+                          vcf_reserved_fields.FORMAT_FIELDS.get(field.name),
+                          formats)
+    else:
+      formats.update({
+          field.name: _Format(field.name,
+                              parser.field_counts[vcfio.MISSING_FIELD_VALUE],
+                              _BIG_QUERY_TYPE_TO_VCF_TYPE_MAP.get(field.type),
+                              field.description)})
+
+
+def _add_info_fields(field, infos):
+  # type: (bigquery.TableFieldSchema, Dict[str, _Info]) -> None
+  if field.name == bigquery_util.ColumnKeyConstants.ALTERNATE_BASES:
+    _add_info_fields_from_alternate_bases(field, infos)
+  else:
+    if field.name in vcf_reserved_fields.INFO_FIELDS.keys():
+      _add_reserved_field(field,
+                          vcf_reserved_fields.INFO_FIELDS.get(field.name),
+                          infos)
+    else:
+      infos.update({
+          field.name: _Info(field.name,
                             parser.field_counts[vcfio.MISSING_FIELD_VALUE],
                             _BIG_QUERY_TYPE_TO_VCF_TYPE_MAP.get(field.type),
-                            field.description)})
+                            field.description,
+                            None,
+                            None)})
 
 
 def _add_info_fields_from_alternate_bases(schema, infos):
   # type: (bigquery.TableFieldSchema, Dict[str, _Info]) -> None
   for field in schema.fields:
     if field.name in _CONSTANT_ALTERNATE_BASES_FIELDS:
       continue
+    if field.name in vcf_reserved_fields.INFO_FIELDS.keys():
+      _add_reserved_field(field,
+                          vcf_reserved_fields.INFO_FIELDS.get(field.name),
+                          infos)
+    else:
+      infos.update({
+          field.name: _Info(field.name,
+                            parser.field_counts[
+                                vcfio.FIELD_COUNT_ALTERNATE_ALLELE],
+                            _BIG_QUERY_TYPE_TO_VCF_TYPE_MAP.get(field.type),
+                            field.description,
+                            None,
+                            None)})
 
-    infos.update({
-        field.name: _Info(field.name,
-                          parser.field_counts[
-                              vcfio.FIELD_COUNT_ALTERNATE_ALLELE],
-                          _BIG_QUERY_TYPE_TO_VCF_TYPE_MAP.get(field.type),
-                          field.description,
-                          None,
-                          None)})
+
+def _add_reserved_field(
+    field_schema,  # type: bigquery.TableFieldSchema
+    reserved_definition,  # type: Union[_Format, _Info]
+    header_fields_map  # type: Dict[str, Union[_Format, _Info]]
+):
+  # type: (...) -> None
+  """Adds the reserved field to `header_fields_map`.
+
+  Raises:
+    ValueError: If the field schema type is not the same as the reserved type.
```

Comment:
Please add a TODO to override this check in case the user does want to have an 'invalid' VCF file anyway.